instrumentation/view-click/src/main/kotlin/io/opentelemetry/android/instrumentation/view/click/ViewClickEventGenerator.kt

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.instrumentation.view.click
+
+import android.view.MotionEvent
+import android.view.View
+import android.view.ViewGroup
+import android.view.Window
+import io.opentelemetry.android.instrumentation.view.click.internal.APP_SCREEN_CLICK_EVENT_NAME
+import io.opentelemetry.android.instrumentation.view.click.internal.VIEW_CLICK_EVENT_NAME
+import io.opentelemetry.android.instrumentation.view.click.internal.viewIdAttr
+import io.opentelemetry.android.instrumentation.view.click.internal.viewNameAttr
+import io.opentelemetry.android.instrumentation.view.click.internal.xCoordinateAttr
+import io.opentelemetry.android.instrumentation.view.click.internal.yCoordinateAttr
+import io.opentelemetry.api.common.Attributes
+import io.opentelemetry.api.incubator.logs.ExtendedLogRecordBuilder
+import io.opentelemetry.api.incubator.logs.ExtendedLogger
+import java.lang.ref.WeakReference
+import java.util.LinkedList
+
+class ViewClickEventGenerator(
+    private val eventLogger: ExtendedLogger,
+) {
+    private var windowRef: WeakReference<Window>? = null
+
+    private val viewCoordinates = IntArray(2)
+
+    fun startTracking(window: Window) {
+        windowRef = WeakReference(window)
+        val currentCallback = window.callback
+        window.callback = WindowCallbackWrapper(currentCallback, this)
+    }
+
+    fun generateClick(motionEvent: MotionEvent?) {
+        windowRef?.get()?.let { window ->
+            if (motionEvent != null && motionEvent.actionMasked == MotionEvent.ACTION_UP) {
+                createEvent(APP_SCREEN_CLICK_EVENT_NAME)
+                    .setAttribute(yCoordinateAttr, motionEvent.y.toDouble())
+                    .setAttribute(xCoordinateAttr, motionEvent.x.toDouble())
+                    .emit()
+
+                findTargetForTap(window.decorView, motionEvent.x, motionEvent.y)?.let { view ->
+                    createEvent(VIEW_CLICK_EVENT_NAME)
+                        .setAllAttributes(createViewAttributes(view))
+                        .emit()
+                }
+            }
+        }
+    }
+
+    fun stopTracking() {
+        windowRef?.get()?.run {
+            if (callback is WindowCallbackWrapper) {
+                callback = (callback as WindowCallbackWrapper).unwrap()
+            }
+        }
+        windowRef = null
+    }
+
+    private fun createEvent(name: String): ExtendedLogRecordBuilder =
+        eventLogger
+            .logRecordBuilder()
+            .setEventName(name)
+
+    private fun createViewAttributes(view: View): Attributes {
+        val builder = Attributes.builder()
+        builder.put(viewNameAttr, viewToName(view))
+        builder.put(viewIdAttr, view.id.toLong())
+
+        builder.put(xCoordinateAttr, view.x.toDouble())
+        builder.put(yCoordinateAttr, view.y.toDouble())
```

Comment:
should we also call `getLocationInWindow` or `getLocationOnScreen` here?

Reply:
i don't think it's necessary here because we're not doing bound checks here.

Reply:
ok, i'm not sure if `view.x` and `view.y` are relative to the parent or absolute to the window, if its relative to the parent, you'd need `getLocationInWindow` right otherwise the position will be wrong

Reply:
From what i understand from the [doc](https://developer.android.com/reference/android/view/View#getX()), it's absolute.